src/Facets/WhitelistManagerFacet.sol

```diff
@@ -0,0 +1,232 @@
+        emit AddressWhitelisted(_contractAddress);
+    }
+
+    /// @dev Internal function to handle whitelist removal logic
+    function _removeFromWhitelist(address _address) internal {
+        if (!LibAllowList.contractIsAllowed(_address)) return;
+
+        LibAllowList.removeAllowedContract(_address);
+        emit AddressRemoved(_address);
+    }
+
+    /// @dev Internal function to handle function selector approval logic
+    function _setFunctionApproval(bytes4 _selector, bool _approval) internal {
+        bool currentlyApproved = LibAllowList.selectorIsAllowed(_selector);
+
+        if (_approval != currentlyApproved) {
+            if (_approval) {
+                LibAllowList.addAllowedSelector(_selector);
+            } else {
+                LibAllowList.removeAllowedSelector(_selector);
+            }
+            emit FunctionSelectorApprovalChanged(_selector, _approval);
+        }
+    }
+
+    /// Temporary methods for migration ///
+    /// @dev Remove these methods after migration is complete in next facet upgrade.
+    /// @inheritdoc IWhitelistManagerFacet
+    function migrate(
+        bytes4[] calldata _selectorsToRemove,
+        address[] calldata _contractsToAdd,
+        bytes4[] calldata _selectorsToAdd
+    ) external {
+        if (msg.sender != LibDiamond.contractOwner()) {
+            LibAccess.enforceAccessControl();
+        }
+
+        LibAllowList.AllowListStorage storage als = _getAllowListStorage();
+
+        // return early if already migrated
+        if (als.migrated) return;
+
+        // clear old state
+        // reset contractAllowList
+        for (uint256 i = 0; i < als.contracts.length; i++) {
+            address contractAddr = als.contracts[i];
+            als.contractAllowList[contractAddr] = false;
+        }
+
+        // reset selectorAllowList with external selectors array because new selectors array does not exist yet
+        for (uint256 i = 0; i < _selectorsToRemove.length; i++) {
+            bytes4 selector = _selectorsToRemove[i];
+            als.selectorAllowList[selector] = false;
+        }
+
+        // reset contract array
+        delete als.contracts;
+        // clearing selectors is not needed as it new variable
+
+        // whitelist contracts
+        for (uint256 i = 0; i < _contractsToAdd.length; i++) {
+            if (_contractsToAdd[i] == address(this)) {
+                revert CannotAuthoriseSelf();
+            }
+
+            // check for duplicate contracts in _contractsToAdd
+            // this prevents both duplicates and ensures all contracts were properly reset
+            if (LibAllowList.contractIsAllowed(_contractsToAdd[i])) {
+                revert InvalidConfig();
+            }
+
+            LibAllowList.addAllowedContract(_contractsToAdd[i]);
+            emit AddressWhitelisted(_contractsToAdd[i]);
+        }
+
+        // whitelist selectors
+        for (uint256 i = 0; i < _selectorsToAdd.length; i++) {
```

Comment:
not super important here. but since we do this across 50+ networks, we can still save gas, where possible.
Consider using our loop pattern to increase the loop variable in an unchecked block instead of in the loop signature.